pkg/cache/cache_ristretto.go

```diff
@@ -0,0 +1,30 @@
+//go:build !wasm
+// +build !wasm
+
+package cache
+
+import (
+	"github.com/dgraph-io/ristretto"
+)
+
+// NewCache creates a new ristretto cache from the given config.
```

Comment:
Should we stick to the "return structs accept interfaces" mantra here?

Reply:
We can, but it does mean that `NewCache`'s signature will change based on the platform, which may cause issues in certain compilation targets

Reply:
I don't get it. In this case it's always going to be a ristretto cache coming out of `cache_ristretto`. The method should probably be called `NewRistrettoCache`, and it will be totally missing on the wasm arch. If there were a wasm compatible version it would be called `NewHashmapCache` or something, and its version of `wrapped` would be specific to that cache impl.

Reply:
> If there were a wasm compatible version it would be called NewHashmapCache or something

Actually, no. The whole point is that it is `NewCache` regardless of platform, so that when it is invoked outside of the package, the caller does not need to know on which platform the code is operating. Otherwise, it would fail to compile.